### PR TITLE
Added sidekiq "RemoveAssignmentsWorker" #1013

### PR DESCRIPTION
**Pull request by feelfreelinux**
Removed slow request call "WikiCourseEdit"#remove_assignments, by creating new sidekiq worker. + improved setup docs.

### CODE REVIEW

File: docs/setup.md [hunk 1]
```diff
@@ -82,7 +82,8 @@
 3. **Start Gulp to watch JS and CSS**
 
       $ `gulp`
-
+      - if you are getting error message "Error: EACCES: permission denied, open '/home/rlot/.babel.json'", use:
+      `BABEL_DISABLE_CACHE=1 gulp`
```
Comment (ragesoss) [inline docs/setup.md]: Let's put this in docs/troubleshooting.md (and add a link to that file from the end up setup.md)

Comment (ragesoss) [inline docs/setup.md]: Also, put in 'username' or the like in place of your own username in the example error message.

File: app/workers/remove_assignments_worker.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
```
Comment (ragesoss) [inline app/workers/remove_assignments_worker.rb]: File name and class name should be singular. (This is how they are called, so it's causing the tests to fail.)

Comment (ragesoss) [inline app/workers/remove_assignments_worker.rb]: Remove this file.

**Comment by feelfreelinux** [comment]
Fixed it

File: app/workers/remove_assignment_worker.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+  def self.schedule_edits(course:, editing_user:, assignment:)
+    perform_async(course.id, editing_user.id, assignment.id)
+  end
+
+  def perform(course_id, editing_user_id, assignment_id)
+    course = Course.find(course_id)
+    editing_user = User.find(editing_user_id)
+    assignment = Assignment.find(assignment_id);
+    WikiCourseEdits.new(action: :remove_assignments, course: course, current_user: editing_user, assignment: assignment)
```
Comment (ragesoss) [inline app/workers/remove_assignment_worker.rb]: Should be `:remove_assignment`

This breaks the UsersController spec. Run this to see if it works: `rspec ./spec/controllers/users_controller_spec.rb`

**Pull request merged**